js/src/forum/utils/DiscussionControls.js

```diff
@@ -230,11 +230,8 @@ export default {
       }
 
       return this.delete().then(() => {
-        // If there is a discussion list in the cache, remove this discussion.
-        if (app.cache.discussionList) {
-          app.cache.discussionList.removeDiscussion(this);
-          m.redraw();
-        }
+        app.cache.discussionList.removeDiscussion(this);
+        m.redraw();
```

Comment:
We're being quite inconsistent with where we trigger redraws - sometimes within the state, sometimes in places like this. But I'll bring up that question in the ticket, it's more of a general point of discussion for me.

Reply:
This is one of the places I was referring to in https://github.com/flarum/core/issues/2144#issuecomment-632957701. 

Potentially, I could see `removeDiscussion()`, `addDiscussion()`, `clear()` and even `setParams()` all trigger redraws - that would slightly change behavior in some places.

Reply:
Add to all except setParams: usually, setParams would be followed by refresh, which already triggers redraw.

Reply:
But isn't the whole point of these state objects that triggering write operations on the state automatically results in a redraw, so that all other places that depend on this state (which the caller is not necessarily aware about) are updated as well?

In other words, if the call has to be followed by another call, why not combine the two? (There might be edge cases or semantics I am missing here...)

Reply:
Good point on this one, not sure why I split it up this way. I added in that setParams calls refresh afterwards. Alternatively, we could refactor it into one method so that if params are provided to refresh, it will set them.

Reply:
Done!

Reply:
Put into 1 method.